src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncIteratorTests.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen
+{
+    [CompilerTrait(CompilerFeature.AsyncStreams)]
+    public class CodeGenAsyncIteratorTests : EmitMetadataTestBase
+    {
+        // PROTOTYPE(async-streams): Consider moving this common test code to TestSources.cs
+        private static readonly string s_common = @"
+namespace System.Collections.Generic
+{
+    public interface IAsyncEnumerable<out T>
+    {
+        IAsyncEnumerator<T> GetAsyncEnumerator();
+    }
+
+    public interface IAsyncEnumerator<out T> : System.IAsyncDisposable
+    {
+        System.Threading.Tasks.Task<bool> WaitForNextAsync();
+        T TryGetNext(out bool success);
+    }
+}
+namespace System
+{
+    public interface IAsyncDisposable
+    {
+        System.Threading.Tasks.Task DisposeAsync();
+    }
+}
+
+namespace System.Runtime.CompilerServices
+{
+    public interface IStrongBox<T>
+    {
+        ref T Value { get; }
+    }
+}
+
+namespace System.Threading.Tasks
+{
+    using System.Runtime.CompilerServices;
+    using System.Threading.Tasks.Sources;
+
+    public struct ManualResetValueTaskSourceLogic<TResult>
+    {
+        public ManualResetValueTaskSourceLogic(IStrongBox<ManualResetValueTaskSourceLogic<TResult>> parent)
+        {
+        }
+
+        public short Version
+            => throw null;
+
+        public ValueTaskSourceStatus GetStatus(short token)
+            => throw null;
+
+        public TResult GetResult(short token)
+            => throw null;
+
+        public void Reset()
+            => throw null;
+
+        public void OnCompleted(Action<object> continuation, object state, short token, ValueTaskSourceOnCompletedFlags flags)
+            => throw null;
+
+        public void SetResult(TResult result)
+            => throw null;
+
+        public void SetException(Exception error)
+            => throw null;
+    }
+}
+";
+
+        [Fact]
+        void TestWellKnownMembers()
```

Comment:
Do test methods need to be `public`?

Reply:
Not that I can tell (TDD.Net, TestExplorer). But it doesn't hurt. 
Fixed